core/src/nucleus/mod.rs

```diff
@@ -49,17 +54,31 @@ impl ZomeFnCall {
             // @see https://github.com/holochain/holochain-rust/issues/198
             id: snowflake::ProcessUniqueId::new(),
             zome_name: zome.to_string(),
-            cap_name: capability.to_string(),
+            cap: cap,
             fn_name: function.to_string(),
             parameters: parameters.into(),
         }
     }
 
     pub fn same_fn_as(&self, fn_call: &ZomeFnCall) -> bool {
         self.zome_name == fn_call.zome_name
-            && self.cap_name == fn_call.cap_name
+            && self.cap == fn_call.cap
```

Comment:
Just checking if this is really the right equality check, I don't know enough to say if it is or not.